src/Command/CodeStudio/CodeStudioChangePHPVersionCommand.php

```diff
@@ -0,0 +1,102 @@
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+
+class CodeStudioChangePHPVersionCommand extends CommandBase {
+
+  use CodeStudioCommandTrait;
+
+  protected static $defaultName = 'codestudio:update-php-version';
+
+  /**
+   * {inheritdoc}.
+   */
+  protected function configure(): void {
+    $this->setDescription('Change the PHP version in Code Studio')
+      ->addArgument('php-version', InputArgument::REQUIRED, 'The PHP version that needs to configured or updated')
+      ->addOption('gitlab-token', NULL, InputOption::VALUE_REQUIRED, 'The GitLab personal access token that will be used to communicate with the GitLab instance')
+      ->addOption('gitlab-host-name', NULL, InputOption::VALUE_REQUIRED, 'The GitLab hostname.');
+    $this->acceptApplicationUuid();
+  }
+
+  /**
+   * @param \Symfony\Component\Console\Input\InputInterface $input
+   * @param \Symfony\Component\Console\Output\OutputInterface $output
+   *
+   * @return int
+   * @throws \Exception
+   */
+  protected function execute(InputInterface $input, OutputInterface $output): int {
+    $php_version = $input->getArgument('php-version');
+    $this->authenticateWithGitLab();
+    $acquiaCloudAppId = $input->getArgument('applicationUuid');
+
+    // @todo: To check if better permission or something available.
+    $ownedProjects = $this->gitLabClient->projects()->all(['owned' => TRUE]);
+
+    // Filter only project where CI/CD is enabled.
+    $ownedProjects = array_filter($ownedProjects, function ($ownedProject) {
+      return $ownedProject['jobs_enabled'];
+    });
+
+    if (empty($ownedProjects)) {
+      $this->io->error('There are no projects available in Code Studio.');
+      return 1;
+    }
+
+    // Flag that will contain the value of the project id to be updated.
+    $project_id = 0;
+    $php_version_already_set = FALSE;
+    foreach ($ownedProjects as $project) {
+      // Get all the variables set for the project.
+      $all_project_variables = $this->gitLabClient->projects()->variables($project['id']);
+      if (!empty($all_project_variables)) {
+        $variables = array_column($all_project_variables, 'value', 'key');
+        // Assuming this project is attached with Acquia cloud only when
+        // ACQUIA_APPLICATION_UUID variable is available/configured.
+        if (!empty($variables['ACQUIA_APPLICATION_UUID']) && $variables['ACQUIA_APPLICATION_UUID'] === $acquiaCloudAppId) {
+          $project_id = $project['id'];
+          $php_version_already_set = $variables['PHP_VERSION'] ?? FALSE;
+          break;
+        }
+      }
+    }
+
+    // If no project found matching with attached Acquia CLoud App id.
+    if (empty($project_id)) {
+      $this->io->error('There are no Acquia Cloud projects available in Code Studio.');
+      return 1;
+    }
+
+    try {
+      // If PHP version is not set in variables.
+      if (!$php_version_already_set) {
+        if ($php_version === '8.0') {
```

Comment:
Will this _always_ be the default? I doubt it. I'd rather not hardcode this message and have it become stale when the default version changes.

Reply:
removed this.